core/src/test/java/org/apache/iceberg/TestIncrementalDataTableScan.java

```diff
@@ -196,56 +193,66 @@ public void testRollbacks() {
     transaction.commitTransaction();
     // Go back to snapshot "E"
     table.manageSnapshots().rollbackTo(5).commit();
-    Assert.assertEquals(5, table.currentSnapshot().snapshotId());
+    assertThat(table.currentSnapshot().snapshotId()).isEqualTo(5);
     filesMatch(Lists.newArrayList("B", "D", "E"), appendsBetweenScan(1, 5));
     filesMatch(Lists.newArrayList("B", "D", "E"), appendsAfterScan(1));
   }
 
-  @Test
+  @TestTemplate
   public void testIgnoreResiduals() throws IOException {
     add(table.newAppend(), files("A"));
     add(table.newAppend(), files("B"));
     add(table.newAppend(), files("C"));
 
-    TableScan scan1 = table.newScan().filter(Expressions.equal("id", 5)).appendsBetween(1, 3);
+    IncrementalAppendScan scan1 =
+        table
+            .newIncrementalAppendScan()
```

Comment:
why has this been changed?

Reply:
`table.newScan(),appendsBetween` is marked as "deprecated" in https://iceberg.apache.org/javadoc/latest/org/apache/iceberg/TableScan.html so that I changed here.
> Deprecated.
since 1.0.0, will be removed in 2.0.0; use [Table.newIncrementalAppendScan()](https://iceberg.apache.org/javadoc/latest/org/apache/iceberg/Table.html#newIncrementalAppendScan()) instead.

But maybe this should be another PR, or keep it this time. Let me revert it to the previous version.